indexing-hadoop/src/main/java/io/druid/indexer/path/PathSpec.java

```diff
@@ -28,13 +28,14 @@
 
 /**
  */
-@JsonTypeInfo(use= JsonTypeInfo.Id.NAME, property="type")
-@JsonSubTypes(value={
-    @JsonSubTypes.Type(name="granular_unprocessed", value=GranularUnprocessedPathSpec.class),
-    @JsonSubTypes.Type(name="granularity", value=GranularityPathSpec.class),
-    @JsonSubTypes.Type(name="static", value=StaticPathSpec.class),
-    @JsonSubTypes.Type(name="dataSource", value=DatasourcePathSpec.class),
-    @JsonSubTypes.Type(name="multi", value=MultiplePathSpec.class)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name= "granular_unprocessed", value = GranularUnprocessedPathSpec.class),
+    @JsonSubTypes.Type(name= "granularity", value = GranularityPathSpec.class),
+    @JsonSubTypes.Type(name= "static", value = StaticPathSpec.class),
+    @JsonSubTypes.Type(name= "dataSource", value = DatasourcePathSpec.class),
+    @JsonSubTypes.Type(name= "multi", value = MultiplePathSpec.class),
+    @JsonSubTypes.Type(name= "partition", value = PartitionPathSpec.class)
```

Comment:
How do you feel about a "recursive" flag in the static path spec? When the flag is set to true, we try to find all the files under a given directory. I think it'll clean up the code even further